mbed_tools_ci_scripts/utils/filesystem_helpers.py

```diff
@@ -119,14 +119,17 @@ def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
         """Context manager exit point."""
         self.cleanup()
 
+    def _cleanup_using_lib_facility(self) -> None:
+        try:
+            self._tmp_dir_context_manager.cleanup()
+        except FileNotFoundError as e:
+            logger.warning(
+                f"Failed cleaning up {self.path}. Reason: {str(e)}")
+
     def cleanup(self) -> None:

```

Comment:
I find it odd that in non-windows case this calls `self.something_else()` and in windows case it does the work here. My gut feeling is this should either do all the work:
```
if platform.system() == 'Windows':
    shutil.rmtree(str(self.path), ignore_errors=True)
else:
    self._tmp_dir_context_manager.cleanup()
```

Or, if you insist on doing `try/except`, delegate always:

```
if platform.system() == 'Windows':
    self._windows_cleanup()
else:
    self._regular_cleanup() # needs a better name
```

Reply:
done